numpy/lib/function_base.py

```diff
@@ -1856,13 +1856,26 @@ def diff(a, n=1, axis=-1):
         The number of times values are differenced.
     axis : int, optional
         The axis along which the difference is taken, default is the last axis.
+    prepend : array_like
+        Values to insert along axis before the differenced results.  'first' is
+        an alias for a.take([0], axis), which inserts the first value from a 
```

Comment:
Although the recipe would actually be `append = a.take([-1], axis) if a.shape[axis] != 0 else None`, which is a bit more verbose